sdk/go/client.go

```diff
@@ -50,33 +50,48 @@ func (fc *GrpcClient) GetOnlineFeatures(ctx context.Context, req *OnlineFeatures
 		return nil, err
 	}
 	resp, err := fc.cli.GetOnlineFeatures(ctx, featuresRequest)
+	r := &OnlineFeaturesResponse{RawResponse: resp}
+	if err != nil {
+		return nil, err
+	}
 
-	// collect unqiue entity refs from entity rows
-	entityRefs := make(map[string]struct{})
-	for _, entityRows := range req.Entities {
-		for ref := range entityRows {
-			entityRefs[ref] = struct{}{}
-		}
+	err = r.stripProjects(req.Entities)
+	if err != nil {
+		return nil, err
 	}
 
-	// strip projects from to projects
-	for _, fieldValue := range resp.GetFieldValues() {
-		stripFields := make(map[string]*types.Value, len(fieldValue.Fields))
-		for refStr, value := range fieldValue.Fields {
-			_, isEntity := entityRefs[refStr]
-			if !isEntity { // is feature ref
-				featureRef, err := parseFeatureRef(refStr, true)
-				if err != nil {
-					return nil, err
+	return r, err
+}
+
+// Wait for and get online feature data from serving.
+// Polls serving for feature data with an exponential backoff if features are not found.
+func (fc *GrpcClient) WaitGetOnlineFeatures(ctx context.Context, req *OnlineFeaturesRequest, waitForSecs float64) (
```

Comment:
I am wondering if this shouldn't be outside of Feast. Is there a reason why whatever is wrapping the client couldn't retry?

Reply:
No. The point here is to abstract the quirk how Feast works, remove need for users to retry so that they can call one method and just get features, instead having to worry about writing a retry loop to wait for feature data because Feast ingestion is eventually consistent.

Reply:
Removed.